### PR TITLE
feat(authentication): Add Generate Challenge 

### PR DESCRIPTION
**Pull request by chanjant**
**NOTE: do not merge until [Prove Ownership Epic](https://www.notion.so/immersve/EPIC-240) is ready to ship**

- Add generate challenge to prove ownership
- Rename existing `Generate Challenge` to `Generate SIWE Challenge for Login` to avoid duplication

Ticket Link: https://www.notion.so/immersve/API-docs-88eb864f76fe4e03beee978a55be78f6?pvs=4

Test deploy url: https://229-merge--jovial-scone-ec740d.netlify.app/api-reference/generate-challenge/

### CODE REVIEW

File: imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  expiresAt:
+    description: The time the challenge expires.
+    type: string
+    example: "2024-03-20T08:20:00.000Z"
+  createdAt:
+    description: The time the challenge was created.
+    type: string
+    example: "2024-03-20T08:20:00.000Z"
+  message:
+    description: The message displayed to the web3 address owner to request their signature.
```
Comment (ncjones) [inline imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml:36]:
```suggestion
    description: The message to be signed by the web3 address owner.
```

File: imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  purpose:
+    description: The purpose of the challenge.
+    type: string
+    example: "claim-web3-address"
+  address:
+    description: The web3 address to which the challenge is directed.
+    type: string
+    example: "0x123..."
+  network:
+    description: The blockchain network associated with the web3 address.
```
Comment (ncjones) [inline imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml:20]: What are the valid values?

Comment (tonykha123) [reply imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml:20]: we should have an enum of the available networks for this @NicBathgate

File: imsv-docs-docusaurus/openapi/endpoints/login/create-challenge.yaml
Comment (tonykha123) [inline imsv-docs-docusaurus/openapi/endpoints/login/create-challenge.yaml:1]: howcome this file is in the login folder?

File: imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+required:
+  - purpose
+  - address
+  - network
+properties:
+  id:
+    description: The unique identifier of the challenge.
+    type: string
+    example: "GlSDzZHcxrSDeJbKu"
+  purpose:
```
Comment (tonykha123) [inline imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml:11]: we should probably also have an enum of supported purposes

File: imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml (hunk 2 above)
Comment (NicBathgate) [reply imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml:20]: At the moment:
      - polygon-mainnet
      - polygon-mumbai
I've pushed a change to enum those two possible values, we will need to make sure we keep this up to date when we add new chains though (including when we replace mumbai with amoy soon)

File: imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml (hunk 3 above)
Comment (NicBathgate) [reply imsv-docs-docusaurus/openapi/endpoints/login/models/create-challenge-response.yaml:11]: Done 👍

File: imsv-docs-docusaurus/openapi/endpoints/login/create-challenge.yaml
Comment (NicBathgate) [reply imsv-docs-docusaurus/openapi/endpoints/login/create-challenge.yaml:1]: @tonykha123 good question. We could create a brand new section for it, I don't think it really fits under any of the others specifically, it's not intended to be exclusively for use with creating funding sources in the future so I didn't move it into there. The section it's in is labeled 'Authentication' which makes more sense than 'login', we could rename the login folder to 'authentication' to match the UI if you think that would be worth it?
CC @ncjones thoughts?

**Comment by NicBathgate** [comment]
Closing as this PR has been merged into Roman's new PR here:
https://github.com/immersve/immersve-docs/pull/234

**Pull request closed without merging**